exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,201 @@
+  {
+    return this->M <= enemyM;
+  }
+  bool compareO(short enemyO) // false/0 = defender, true/1 = attacker
+  {
+    return this->O <= enemyO;
+  }
+  friend void swap();
+  short const getN() { return N; };
+  short const getM() { return M; };
+  short const getO() { return O; };
+};
+
+void swap(short &a, short &b)
+{
+  short temp = a;
+  a = b;
+  b = temp;
+}
+
+bool coin_flip()
+{
+  // false/0 = defender, true/1 = attacker
+  string choice;
+  do
+  {
+    cout << "Heads or tails?" << endl;
+    cin >> choice;
+    for (int i = 0; i < choice.length(); i++) // formatting the input, just in case someone types something like "TaIlS"
+      choice[i] = tolower(choice[i]);
+
+    bool coin = rand() % 2; // heads = 0, tails = 1
+    cout << endl;
+    if (coin == 0)
+      cout << "It's heads!" << endl;
+    else
+      cout << "It's tails!" << endl;
+
+    if (choice == "heads" && coin == 0)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    if (choice == "tails" && coin == 1)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    if (choice != "heads" && choice != "tails")
+    {
+      cout << "But you wrote neither heads nor tails... try again." << endl;
+      continue;
+    }
+    cout << "You're the defender! (Blue)" << endl
+         << endl;
+    return false;
+  } while (true);
+}
+
+int main()
+{
+  srand(time(0));
+  do
+  {
+    short scoreRed = 0, scoreBlue = 0; // players scores
+    PlayerDices Red;
+    PlayerDices Blue;
+    bool priority = coin_flip();
+    cout << "Red dices:" << endl;
+    Red.tripleDiceThrow();
+    cout << endl;
+    cout << "Blue dices:" << endl;
+    Blue.tripleDiceThrow();
+    cout << endl;
+
+    cout << "  R    B" << endl; // false/0 = defender, true/1 = attacker
+    // super cool graphics ahead! wowowowow! vvv
+    if (Red.compareN(Blue.getN()))
+    {
+      cout << "N " << Red.getN() << " vs " << Blue.getN() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "N " << Red.getN() << " vs " << Blue.getN() << " => red wins!" << endl;
+      scoreRed++;
+    }
+    if (Red.compareM(Blue.getM()))
+    {
+      cout << "M " << Red.getM() << " vs " << Blue.getM() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "M " << Red.getM() << " vs " << Blue.getM() << " => red wins!" << endl;
+      scoreRed++;
+    }
+    if (Red.compareO(Blue.getO()))
+    {
+      cout << "O " << Red.getO() << " vs " << Blue.getO() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "O " << Red.getO() << " vs " << Blue.getO() << " => red wins!" << endl;
+      scoreRed++;
+    }
+
+    (scoreBlue > scoreRed) ? cout << "BLUE WINS! (" << scoreBlue << " to " << scoreRed << ")" << endl : cout << "RED WINS! (" << scoreRed << " to " << scoreBlue << ")" << endl;
```

Comment:
è una condizione ternaria un po' lunga, forse in questi casi sarebbe meglio if ed else

Reply:
sistemato!